### PR TITLE
Vault in kitt

### PR DESCRIPTION
**Pull request by jojomomojo**
Vault is now in kitt, using consul as the backend data store

No more kitt.run -- bring your own domain name and set KITT_DOMAIN

DNS setup:

traefik-kitt.yourdomain A 169.254.32.1
consul.yourdomain A 169.254.32.1
vault.yourdomain A 169.254.32.1

and you can optionally set *.yourdomain to 169.254.32.1 to default to local services, and CNAME services backed by Argo tunnel to traefik.yourdomain (enable Proxy).  Or just do the CNAME part.  Up to the domain owner how they want to use those names.

### CODE REVIEW

File: scripts/unseal.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from status import status
+
+SHELL = '/usr/bin/env bash'
+
+# curl --request PUT --data '{"key": "abcd1234..."}' https://vault.$KITT_DOMAIN/v1/sys/unseal
+def unseal():
+    print('Unseal Moria')
+    url = "vault.{}".format(os.getenv("KITT_DOMAIN"))
+    for index in range(1, 4):
+        key = local.shell(f'pass moria/keys_{index} | base64 -d')
```
Comment (bellaweo) [inline scripts/unseal.py:17]: i'm on linux and get this error when running `make unseal`

```
$ make unseal
--> Loading config...
--> Loading inventory...

--> Connecting to hosts...
    [@local] Connected

--> Preparing operations...
    Loading: scripts/unseal.py
Unseal Moria
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 854, in gevent._gevent_cgreenlet.Greenlet.run
  File "/home/mel/.local/lib/python3.8/site-packages/pyinfra/api/util.py", line 402, in read_buffer
    line = line.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xaf in position 1: invalid start byte
2020-07-15T14:56:19Z <Greenlet at 0x7f9a72581ae0: read_buffer('stdout', <FileObjectPosix at 0x7f9a73f89400 _fobj=<_io.Buff, <Queue at 0x7f9a724a8d00 queue=deque([('stderr', ', print_output=False, print_func=<function read_buffers_into_queue.<locals>.<lambda)> failed with UnicodeDecodeError

--> [@local] pyinfra error: Local command failed: pass moria/keys_1 | base64 -d
['base64: invalid input']
make: *** [Makefile:70: unseal] Error 1

```

i'm pretty sure the pass moria/key_x vars aren't base64 encoded.

and on line 19 below, the data for that put request should look something like {"key": "1234" }

Comment (bellaweo) [reply scripts/unseal.py:17]: this diff works for me:

```
diff --git a/scripts/unseal.py b/scripts/unseal.py
index 14ff70f..21fb6e8 100644
--- a/scripts/unseal.py
+++ b/scripts/unseal.py
@@ -14,9 +14,9 @@ def unseal():
     print('Unseal Moria')
     url = "vault.{}".format(os.getenv("KITT_DOMAIN"))
     for index in range(1, 4):
-        key = local.shell(f'pass moria/keys_{index} | base64 -d')
+        key = local.shell(f'pass moria/keys_{index}')
         try:
-            req = requests.put(f'https://{url}/v1/sys/unseal', data = f'{key}')
+            req = requests.put(f'https://{url}/v1/sys/unseal', data = '{{"key": "{}"}}'.format(key))
             req.raise_for_status()
         except requests.exceptions.RequestException as e:
             raise SystemExit(e)
```

Comment (jojomomojo) [reply scripts/unseal.py:17]: @bellaweo can you change it back to what works for you? I'll figure out why I had to make those changes.  What python are you running, 2 or 3?

Comment (bellaweo) [reply scripts/unseal.py:17]: Python 3.8.2

File: docker-compose.yml [hunk 2]
```diff
@@ -136,6 +138,50 @@ services:
...
+      - AWS_ACCESS_KEY_ID
+      - AWS_SECRET_ACCESS_KEY
+      - AWS_SECURITY_TOKEN
+      - AWS_SESSION_TOKEN
+    volumes:
+      - ./etc/vault:/config
+      - ./vault-service:/service
+    ports:
+      - 169.254.32.1:8200:8200
+      - 169.254.32.1:8500:8500
```
[the remark this answers is not in the document]
Comment (jojomomojo) [reply docker-compose.yml:160]: dupe was on vault service, copied from moria.  fixed

**Pull request merged**